src/core/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { Component, forwardRef } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../theme';
+import { baseStyles } from './Pagination.baseStyles';
+import { HtmlSpan, HtmlNav, HtmlDiv } from '../../reset';
+import { PageInput, PageInputValue } from './PageInput/PageInput';
+import { Button } from '../Button/Button';
+import { VisuallyHidden } from '../VisuallyHidden/VisuallyHidden';
+import { AutoId } from '../utils/AutoId/AutoId';
+
+export interface PageInputProps {
+  /** Page input action button label for screen readers  */
+  buttonText: string;
+  /** Input placeholder text */
+  inputPlaceholderText: string;
+  /** Error text shown with invalid page number input */
+  invalidValueErrorText: (value: PageInputValue) => string;
+  /** Visually hidden label for input field */
+  labelText: string;
+}
+
+type ShowInputProps =
+  | {
+      pageInput: false;
+      pageInputProps?: never;
+    }
+  | {
+      /** Show input field for page number */
+      pageInput?: true;
+      /** Props for page input field */
+      pageInputProps: PageInputProps;
+    };
+
+interface InternalPaginationProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** aria-label for pagination  */
+  'aria-label': string;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated automatically
+   */
+  id?: string;
+  /** Function for page number indicator text */
+  pageIndicatorText?: (currentPage: number, lastPage: number) => string;
```

Comment:
Should pageIndicatorText be required since else it will be blank?

Reply:
I suppose there is no reason to make it easy to use this without texts. Changed now.